datadog/resource_datadog_monitor.go

```diff
@@ -342,6 +349,22 @@ func resourceDatadogMonitorExists(d *schema.ResourceData, meta interface{}) (b b
 	return true, nil
 }
 
+func resourceDatadogMonitorCustomizeDiff(diff *schema.ResourceDiff, meta interface{}) error {
+	if _, ok := diff.GetOk("query"); !ok {
+		// If query depends on previous resources, we can't validate
```

Comment:
Suggestion on this log line based on our offline chat that clarifies things for me :) 
```suggestion
		// If "query" depends on other resources, we can't validate as the variables may not be interpolated yet. 
```